### PR TITLE
Replace deprecated loaders with new `assets-module`

### PR DESCRIPTION
**Pull request by dagda1**
## Hey, I just made a Pull Request!

We encountered a bug using webpack in `@backstage/cli` and it is because the `file-loader` no longer seems to work with fonts.

The file loader has actually been deprecated in favour of [assets-module](https://webpack.js.org/guides/asset-modules/).

It turns out that the `file-loader`, `url-loader` and `raw-loader` have all been deprecated but this PR just raises the point and fixes an actual bug with fonts.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/curvy-timers-sit.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/cli': minor
```
Comment (Rugvip) [inline .changeset/curvy-timers-sit.md:2]: Should be a `patch` as it's not breaking and we're on 0.x

File: packages/cli/src/lib/bundler/transforms.ts [hunk 2]
```diff
@@ -90,15 +90,11 @@ export const transforms = (options: TransformOptions): Transforms => {
...
-          loader: require.resolve('file-loader'),
-          options: {
-            name: 'static/[name].[hash:8].[ext]',
-          },
-        },
-      ],
+      test: /\.(eot|woff|woff2|ttf)$/i,
+      type: 'asset/resource',
+      generator: {
+        filename: 'static/[hash][ext][query]',
```
Comment (Rugvip) [inline packages/cli/src/lib/bundler/transforms.ts:96]: I would like to keep the `[name]` prefix in there just to make it a bit easier to navigate the output and discover problems. Curious also about the query, do you know why it's in there? I see it's default

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

Should definitely switch to asset modules for all the assets. I'm curious, what is the bug that you encountered with the current setup?

File: packages/cli/src/lib/bundler/transforms.ts (hunk 2 above)
Comment (dagda1) [reply packages/cli/src/lib/bundler/transforms.ts:96]: I wondered that too, I think it is for the case where a loader might have a query string that you sometimes see.

Comment (dagda1) [reply packages/cli/src/lib/bundler/transforms.ts:96]: > Thank you! 👍
> 
> Should definitely switch to asset modules for all the assets. I'm curious, what is the bug that you encountered with the current setup?

You need an actual font-file which in my case was a .ttf looked like this

```ts
@font-face {
  font-family: 'GeosansLight';
  src: url('../fonts/GeosansLight/GeosansLight.ttf');
}
```

You get this error

> Should not import the named export 'version'.'split' (imported as 'version') from default-exporting module (only default export is available soon)

Not the most helpful error but changing the `file-loader` for the assets module fixed it.

**Comment by benjdlambert** [comment]
@dagda1 sorry to piggyback on the back of this PR, but I went ahead and moved the other ones across too, just need to do some final testing tomorrow before we ship this, but i'm out for the day!

**Comment by dagda1** [comment]
> @dagda1 sorry to piggyback on the back of this PR, but I went ahead and moved the other ones across too, just need to do some final testing tomorrow before we ship this, but i'm out for the day!

totally fine by me

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**